### PR TITLE
Normalize quotes in CSS like in JS

### PR DESCRIPTION
**Pull request by lydell**
This enforces single/double quotes like we do in JS but for CSS (in all parts of CSS I could think of).

Related to this is handling of escapes. In JS, we don't normalize escapes in strings in any way except removing unnecessary escapes. In CSS, I think we should simply leave all escapes alone:

- Escapes in CSS are very uncommon, so IMO it is not worth spending complexity on them.
- PostCSS plugins (or preprocessors) could translate the otherwise unnecessary escape "\n" into "\A", for example.
- Stylelint does not seem to have escaping rules.
- Escapes can also appear in IDs and stuff (not just in strings), complicating things furhter.

### CODE REVIEW

File: tests/css_quotes/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    content: "He's sayin': \\"How's it goin'?\\" Don't ask me why.";
+    content: "He's sayin': \\"How's it goin'?\\" Don't ask me why.";
+
+    /* Somewhat more real-word example 2. */
+    content: 'var backslash = "\\\\", doubleQuote = \\'"\\';';
+    content: 'var backslash = "\\\\", doubleQuote = \\'"\\';';
+
+    /* Leave all "escapes" alone. */
+    content: '\\Abc4 foo \\n' '\\end';
+    content: '\\Abc4 foo \\n' '\\end';
```
Comment (lydell) [inline tests/css_quotes/__snapshots__/jsfmt.spec.js.snap]: The comments between the strings seem to be missing here, but should not be because of this PR.

**Comment by lydell** [comment]
Could somebody help me understand what's going on with the Travis tests?

**Comment by maxdeviant** [comment]
@lydell It appears that some of the tests are failing, starting [here](https://travis-ci.org/prettier/prettier/jobs/264315927#L502).

**Comment by lydell** [comment]
@maxdeviant Thanks but I understood that too.

Luckily, I figured it out. I forgot that the tests run with `AST_COMPARE=1` on Travis. That’s why they fail there and not when I ran the tests locally. I need to update AST_COMPARE to handle the quote differences in CSS.

**Comment by lydell** [comment]
So the problem here is that `.value` on several CSS AST nodes contains the quotes themselves, so AST_COMPARE=1 thinks that the AST has changed. It feels like I have to adjust `massageAST` to normalize that difference away, but I can’t think of the correct way of doing this yet ...

**Comment by vjeux** [comment]
Feel free to a bit lose on massageAst and just delete the value field if it contains a \.

It's not going to catch those but will let you move forward

**Comment by vjeux** [comment]
contains a backslash

File: tests/css_quotes/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (vjeux) [inline tests/css_quotes/__snapshots__/jsfmt.spec.js.snap]: Known issue: https://github.com/prettier/prettier/issues/2127

Would be awesome to fix this one as well. Shouldn't be hard: make a tokenizer that extracts out comments from a css file and add it through prettier comment location placement logic.

**Comment by vjeux** [comment]
Awesome! As soon as the tests pass, it should be good to merge :)

**Comment by lydell** [comment]
The tests pass now. Is my solution OK?

**Pull request merged**